File: MoviesShared/sharedLogic/src/commonMain/kotlin/com/aryanrogye/movies_shared/models/TVShow.kt
package com.aryanrogye.movies_shared.models

import kotlinx.serialization.Serializable

@Serializable
data class KTTVShow(
    val id: Int,
    val name: String,
    val overview: String,
    val backdropPath: String? = null,
    val numberOfEpisodes: Int,
    val numberOfSeasons: Int,
    val seasons: List<KTSeason>
)

@Serializable
data class KTSeason(
    val id: Int,
    val name: String,
    val seasonNumber: Int,
    val episodeCount: Int,
    val posterPath: String? = null,

)

File: MoviesAPIClient/Features/TMDB/Models/TV/TVShow.swift
//
//  TVShow.swift
//  MoviesAPIClient
//
//  Created by Aryan Rogye on 9/14/26.
//

import Foundation

struct TVShow: Decodable {
    let id: Int
    let name: String
    let overview: String
    let numberOfEpisodes: Int
    let numberOfSeasons: Int
    let seasons: [Season]
}

struct Season: Decodable, Identifiable, Equatable {
    let id: Int
    let name: String
    let seasonNumber: Int
    let episodeCount: Int
    let posterPath: String?
}
